torch/csrc/jit/codegen/cuda/scheduler/mma_utils.cpp

```diff
@@ -63,6 +88,23 @@ IterDomain* getMmaOperandRootDimension(
   if (isVolta(options.macro)) {
     return getMmaOperandRootDimension3d(
         tv, options.operand_layout, mma_dimension);
+  } else if (isTuring(options.macro)) {
+    // Volta mma swizzle requires the broadcast dimension to
+    //  participate, which is not true in Turing+. So the two
+    //  cases w/ or w/o the broadcast are supported here for
+    //  mma pre-swizzle validation.
```

Comment:
This one is about which tensor we place in shared mem and is related to the details of mma format.

Say for example the operand A in TT, and we have this pattern `input->broadcast->reduction`:

When we swizzle for mma format we need the input in shared mem. 

If the shared mem is placed before the broadcast, A looks like A[I, I], 
if after A looks like [I, B, I].

The volta swizzle **requires** that the broadcast axis is present, kind of the only remaining difference in volta mma in the current approach (what the comment meant by broadcast "participating" in the swizzle). volta mma doesn't use ldmatrix but it is through unified user interface. 
 
Turing mma's are either way. So in using `ldmatrix` for turing mma's the utility function supports both cases so that user can decide which way fits better. 

On ampere it is the other way if we use cp async, since we go straight from gmem to smem, so A has to be A[I, I] because there's not broadcast op in between. 

The comment in `ldmatrix` util essentially just states that it is aware that there are two possible input patterns and it's ok either way. 

All of the facts above are actively checked in the mma util.